core/src/main/java/org/bitcoinj/store/DatabaseFullPrunedBlockStore.java

```diff
@@ -619,7 +619,9 @@ protected void putUpdateStoredBlock(StoredBlock storedBlock, boolean wasUndoable
             s.setBytes(1, hashBytes);
             s.setBytes(2, storedBlock.getChainWork().toByteArray());
             s.setInt(3, storedBlock.getHeight());
-            s.setBytes(4, storedBlock.getHeader().unsafeBitcoinSerialize());
+            s.setBytes(4, storedBlock.getHeader().getTransactions() == null ?
```

Comment:
Ditto. This is a full store: getTransactions() should never be null, I think? So it needs to always be run through cloneAsHeader()